scipy/stats/_quantile.py

```diff
@@ -1,24 +1,23 @@
 import numpy as np
 from scipy.special import betainc
-from scipy._lib._array_api import xp_default_dtype, xp_ravel, array_namespace
+from scipy._lib._array_api import xp_ravel, array_namespace, xp_promote
 import scipy._lib.array_api_extra as xpx
 from scipy.stats._axis_nan_policy import _broadcast_arrays, _contains_nan
 from scipy.stats._stats_py import _length_nonmasked
 
 
 def _quantile_iv(x, p, method, axis, nan_policy, keepdims):
     xp = array_namespace(x, p)
-    x = xp.asarray(x)
-    p = xp.asarray(p)
 
-    if not xp.isdtype(x.dtype, ('integral', 'real floating')):
+    if not xp.isdtype(xp.asarray(x).dtype, ('integral', 'real floating')):
         raise ValueError("`x` must have real dtype.")
-    if xp.isdtype(x.dtype, 'integral'):
-        x = xp.astype(x, xp_default_dtype(xp))
 
-    if not xp.isdtype(p.dtype, 'real floating'):
+    if not xp.isdtype(xp.asarray(p).dtype, 'real floating'):
         raise ValueError("`p` must have real floating dtype.")
 
+    x, p = xp_promote(x, p, force_floating=True, xp=xp)
```

Comment:
The comment about this check being very expensive for lists still stands.
I think it should be removed outright.
With the check in, `force_floating=True` is redundant.

Reply:
I just committed your last comment about this line. https://github.com/scipy/scipy/pull/22695#discussion_r2014120797